pkg/preflight/validate.go

```diff
@@ -308,34 +307,31 @@ func deleteTmpPVC(l *log.Logger, client k8sclient.Interface, pvc *corev1.Persist
 			return err
 		}
 	}
-	waitFor = append(waitFor, pvc.Name)
 
-	timeout := time.After(5 * time.Minute)
+	pv, ok := pvsByPVCName[pvc.Name]
+	if !ok {
+		l.Printf("failed to find pv for temp pvc %s", pvc.Name)
+		return nil
+	}
+
+	timeout := time.After(deleteTimeout)
 	interval := time.NewTicker(5 * time.Second)
 	defer interval.Stop()
-	for _, pvc := range waitFor {
-		pv, ok := pvsByPVCName[pvc]
-		if !ok {
-			l.Printf("failed to find pv for temp pvc %s", pvc)
-			continue
+	for {
+		// break the loop as soon as we can't find the pv anymore.
+		if _, err := client.CoreV1().PersistentVolumes().Get(
+			ctx, pv.Name, metav1.GetOptions{},
+		); err != nil && !k8serrors.IsNotFound(err) {
+			l.Printf("failed to get pv for temp pvc %s: %s", pvc, err)
+		} else if err != nil && k8serrors.IsNotFound(err) {
+			break
 		}
 
-		for {
-			// break the loop as soon as we can't find the pv anymore.
-			if _, err := client.CoreV1().PersistentVolumes().Get(
-				ctx, pv.Name, metav1.GetOptions{},
-			); err != nil && !k8serrors.IsNotFound(err) {
-				l.Printf("failed to get pv for temp pvc %s: %s", pvc, err)
-			} else if err != nil && k8serrors.IsNotFound(err) {
-				break
-			}
-
-			select {
-			case <-interval.C:
-				continue
-			case <-timeout:
-				return fmt.Errorf("failed to delete pvs: timeout")
-			}
+		select {
+		case <-interval.C:
+			continue
+		case <-timeout:
+			return fmt.Errorf("failed to delete pvs: timeout")
 		}
 	}
 	return nil
```

Comment:
```suggestion
```

Reply:
I can't remove this return statement, Go requires a `return`.

Reply:
this felt like unreachable code in my head based on the other change requested in my review. but ill take your word for it